cheroot/ssl/builtin.py

```
@@ -172,22 +235,155 @@ def get_environ(self, sock):
         }
 
+        # Python 3.3+
+        if hasattr(sock, 'compression'):
+            compression = sock.compression()
+            if compression is not None:
+                ssl_environ['SSL_COMPRESS_METHOD'] = compression
+
+        # Python 3.6+
+        if hasattr(sock, 'session'):
+            ssl_environ['SSL_SESSION_ID'] = sock.session.id.hex()
+        if hasattr(sock.context, 'get_ciphers'):
+            for cip in sock.context.get_ciphers():
+                if cip['name'] == cipher[0] and cip['protocol'] == cipher[1]:
+                    ssl_environ['SSL_CIPHER_ALGKEYSIZE'] = cip['alg_bits']
+                    break
+
+        # Python 3.7+ sni_callback
+        if hasattr(sock, 'sni') and sock.sni:
+            ssl_environ['SSL_TLS_SNI'] = sock.sni
+
         if self.context and self.context.verify_mode != ssl.CERT_NONE:
             client_cert = sock.getpeercert()
             if client_cert:
-                for cert_key, env_var in self.CERT_KEY_TO_ENV.items():
-                    ssl_environ.update(
-                        self.env_dn_dict(env_var, client_cert.get(cert_key)),
-                    )
+                ssl_environ['SSL_CLIENT_VERIFY'] = 'SUCCESS'
+                ssl_environ.update(
+                    self.env_cert_dict('SSL_CLIENT', client_cert),
+                )
+                ssl_environ['SSL_CLIENT_CERT'] = ssl.DER_cert_to_PEM_cert(
+                    sock.getpeercert(binary_form=True),
+                ).strip()
+
+        ssl_environ.update(self._server_env)
+
+        # not supplied by the Python standard library (as of 3.8)
+        # - SSL_SESSION_RESUMED
+        # - SSL_SECURE_RENEG
+        # - SSL_CLIENT_CERT_CHAIN_n
+        # - SRP_USER
+        # - SRP_USERINFO
 
         return ssl_environ
 
+    def parse_cert(self, path, private_key=False):
+        """Parse a certificate."""
+        # Python 3+ Unix, Python 3.5+ Windows
+        if private_key is not False:
+            try:
```

I'd use `contextlib.suppress` if you need to ignore an exception. This also needs a comment explaining such a decision.

Also, that's a huge nested block, let's save some spaces by factoring it out into a private helper method.

---

done

I also refined the exceptions ignored